src/global/pid/RichTrack_factory.cc

```diff
@@ -1,53 +1,53 @@
-// Copyright 2022, Christopher Dilks
+// Copyright (C) 2022, 2023, Christopher Dilks
 // Subject to the terms in the LICENSE file found in the top-level directory.
 
 #include "RichTrack_factory.h"
 
-#include <edm4eic/vector_utils.h>
-
 //-----------------------------------------------------------------------------
 void eicrecon::RichTrack_factory::Init() {
   auto app = GetApplication();
 
   // input tags
   auto detector_name = eicrecon::str::ReplaceAll(GetPluginName(), ".so", "");
-  auto param_prefix = detector_name + ":" + GetTag();
-  InitDataTags(param_prefix);
-  m_radiatorID = richgeo::ParseRadiatorName(GetTag());
+  auto param_prefix  = detector_name + GetPrefix();
 
   // services
   m_richGeoSvc = app->GetService<RichGeo_service>();
   m_actsSvc    = app->GetService<ACTSGeo_service>();
   InitLogger(param_prefix, "info");
   m_propagation_algo.init(m_actsSvc->actsGeoProvider(), m_log);
-  m_log->debug("detector_name='{}'  param_prefix='{}'  m_radiatorID={}", detector_name, param_prefix, m_radiatorID);
+  m_log->debug("detector_name='{}'  param_prefix='{}'", detector_name, param_prefix);
 
-  // default configuration parameters
-  /* FIXME: eventually we will move this factory to an independent algorithm
-   * and be able to use a Config class, overridable in {D,PF}RICH.cc; for now,
-   * a quick way to set the defaults without reco_flags.py, to tide us over
-   * until we have a proper config file front end
-   */
-  m_numPlanes = 5;
-  if(param_prefix.rfind("DRICH")!=std::string::npos || param_prefix.rfind("PFRICH")!=std::string::npos) {
-    if(param_prefix.rfind("Aerogel")!=std::string::npos) m_numPlanes = 5;
-    else if(param_prefix.rfind("Gas")!=std::string::npos) m_numPlanes = 10;
+  // get list of radiators
+  std::map<int,std::string> radiator_list;
+  for(auto& output_tag : GetOutputTags()) {
+    auto radiator_id = richgeo::ParseRadiatorName(output_tag/*, m_log*/); // FIXME: m_log argument requires PR #608
+    radiator_list.insert({
+        radiator_id,
+        richgeo::RadiatorName(radiator_id/*, m_log*/) // FIXME: m_log argument requires PR #608
+        });
   }
```

Comment:
It would be better to merge #608 first, so we don't have to do this workaround
```suggestion
  for(auto& output_tag : GetOutputTags()) {
    auto radiator_id = richgeo::ParseRadiatorName(output_tag, m_log);
    radiator_list.insert({
        radiator_id,
        richgeo::RadiatorName(radiator_id, m_log)
        });
  }
```